### PR TITLE
chore: remove the openfga JSON file

### PR DESCRIPTION
**Pull request by kian99**
## Description

This PR removes the need to keep a copy of the OpenFGA auth model in DSL format and JSON format, opting only for the DSL since it is more human readable. The 3 areas where we used the JSON have been refactored to obtain it dynamically from the DSL format.
1. In tests, using the `github.com/openfga/language/pkg/go` package.
2. In the ROCK using the `fga` CLI.
3. In the Docker compose using the `fga` CLI.

I built the rock and verified the file was created correct and the docker compose is tested when the integration test is run.

Fixes [JUJU-7187](https://warthogs.atlassian.net/browse/JUJU-7187)

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [ ] Covered by unit tests
- [ ] Covered by integration tests

[JUJU-7187]: https://warthogs.atlassian.net/browse/JUJU-7187?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: rocks/jimm.yaml [hunk 1]
```diff
@@ -39,7 +39,9 @@ parts:
...
 
-            # Copied for the charm to collect and apply, not this container itself
+            # Convert the OpenFGA auth model to JSON and place a copy in the ROCK
+            # Used by the charm to apply to the OpenFGA server, not by the container itself
             mkdir -p $CRAFT_PART_INSTALL/root/openfga/
-            cp ./openfga/authorisation_model.json $CRAFT_PART_INSTALL/root/openfga/authorisation_model.json
-
+            # Note that we can't use go run directly (https://github.com/canonical/rockcraft/issues/755)
+            go install github.com/openfga/cli/cmd/fga@latest
+            /root/go/bin/fga model transform --file ./openfga/authorisation_model.fga --output-format json > $CRAFT_PART_INSTALL/root/openfga/authorisation_model.json
```
Comment (SimoneDutto) [inline rocks/jimm.yaml:47]: why do you need the json file in the first place? The docker compose doesn't use it anymore, and you are transforming the yaml for the backend and openfga entrypoint.
What's missing?

Comment (kian99) [reply rocks/jimm.yaml:47]: The comment above explains why it's in the rock, it's for the charm to retrieve and then set on the OpenFGA server when creating the store.

Comment (SimoneDutto) [reply rocks/jimm.yaml:47]: yes, but isn't the openfga charm able to now being started with the yaml, without worrying about the json?
From the docker compose it seems openfga doesn't need the json anymore.

Comment (kian99) [reply rocks/jimm.yaml:47]: In the docker compose it also needs the json.

Comment (kian99) [reply rocks/jimm.yaml:47]: Look at the changes in entrypoint.sh to see where I convert it.

Comment (SimoneDutto) [reply rocks/jimm.yaml:47]: ah yes, you are right!

**Review by SimoneDutto — APPROVED** [review]
lgtm

**Review by ale8k — APPROVED** [review]
nice thank you!

**Pull request merged**